test/normalized_belongs_to_test.rb

```diff
@@ -60,4 +60,28 @@ def test_fetching_polymorphic_belongs_to_association
 
     assert_equal @parent_record, PolymorphicRecord.first.fetch_owner
   end
+
+  def test_returned_record_should_be_readonly_on_cache_hit
+    IdentityCache.with_fetch_read_only_records do
+      Item.fetch(@parent_record.id) # warm cache
+      assert @record.fetch_item.readonly?
+      assert_equal false, @record.item.readonly?
+    end
+  end
+
+  def test_returned_record_should_be_readonly_on_cache_miss
+    IdentityCache.with_fetch_read_only_records do
+      assert @record.fetch_item.readonly?
+      assert_equal false, @record.item.readonly?
```

Comment:
I usually prefer `assert_equal false, ` over `refute` because it is more specific, however, I don't care as much here since `readonly?` is a rails method.